src/server/pps/server/api_server.go

```diff
@@ -2380,12 +2383,19 @@ func (a *apiServer) CreatePipelineInTransaction(txnCtx *txncontext.TransactionCo
 			return errors.EnsureStack(err)
 		}
 		newPipelineInfo.AuthToken = token
-
 		return nil
 	}(); err != nil {
 		return err
 	}
-
+	// TODO:
+	// - should this be done before the start of the txn?
+	// - should this be done in a retry?
```

Comment:
Yes to both.  What we should really do is make txnCtx some interface that only has `Value(key any) any`, and make the logger stuff take that interface instead of context.Context.  That way we can log in transactions, but not do I/O.